docs/developer_guide/contributing.md

```diff
@@ -0,0 +1,100 @@
+# Contributing
+
+Everybody is welcome to contribute, and we value everybody’s contribution. Code is thus not the only
+way to help the community. Answering questions, helping others, reaching out and improving the
+documentation are immensely valuable contributions as well.
+
+It also helps us if you spread the word: reference the library in blog posts on the awesome
+projects it made possible, shout out on Twitter every time it has helped you, or simply star the
+repo to say "thank you".
+
+Check out the official [ludwig docs](https://ludwig-ai.github.io/ludwig-docs/) to get oriented
+around the codebase, and join the community!
+
+## Open Issues
+
+Issues are listed at: <https://github.com/ludwig-ai/ludwig/issues>
+
+If you would like to work on any of them, make sure it is not already assigned to someone else.
+
+You can self-assign it by commenting on the Issue page with one of the keywords: `#take` or
+`#self-assign`.
+
+Work on your self-assigned issue and eventually create a Pull Request.
+
+## Creating Pull Requests
+
+1. Fork the [repository](https://github.com/ludwig-ai/ludwig) by clicking on the "Fork" button on
+   the repository's page. This creates a copy of the code under your GitHub user account.
+
+2. Clone your fork to your local disk, and add the base repository as a remote:
+
+   ```bash
+   git clone git@github.com:<your Github handle>/ludwig.git
+   cd ludwig
+   git remote add upstream https://github.com/ludwig-ai/ludwig.git
+   ```
+
+3. Create a new branch to hold your development changes:
+
+   ```bash
+   git checkout -b a-descriptive-name-for-my-changes
+   ```
+
+   *Do not*\* work on the `master` branch.
+
+4. Set up a development environment by running the following command in a virtual environment:
+
+   ```bash
+   pip install -e .
+   pip install pre-commit
+   pre-commit install
+   ```
+
+5. Develop features on your branch.
+
+6. Format your code by running pre-commits so that your newly added files look nice:
+
+   ```bash
+   pre-commit run
+   ```
+
+   Pre-commits also run automatically when committing.
+
+7. Once you're happy with your changes, make a commit to record your changes locally:
+
+   ```bash
+   git add .
+   git commit
+   ```
+
+   It is a good idea to sync your copy of the code with the original repository regularly. This
+   way you can quickly account for changes:
+
+   ```bash
+   git fetch upstream
+   git rebase upstream/master
+   ```
+
+   Push the changes to your account using:
+
+   ```bash
+   git push -u origin a-descriptive-name-for-my-changes
+   ```
+
+8. Once you are satisfied, go the webpage of your fork on GitHub. Click on "Pull request" to send
+   your contribution to the project maintainers for review.
```

Comment:
Should we add a 9th step to create a PR to update the docs if it is relevant/a new feature?